### PR TITLE
Add post proposal tests

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Add post proposal tests

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7" target="_blank">Graphite</a> 
      * **PR #9** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/9" target="_blank">Graphite</a> 
        * **PR #11** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11" target="_blank">Graphite</a> 
          * **PR #12** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/12" target="_blank">Graphite</a>  👈
            * **PR #13** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/13" target="_blank">Graphite</a> 
              * **PR #14** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/14" target="_blank">Graphite</a> 
                * **PR #15** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/15" target="_blank">Graphite</a> 
                  * **PR #17** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/17" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/12?utm_source=stack-comment).

File: test/PooltogetherGovernor.t.sol [hunk 1]
```diff
@@ -135,4 +137,162 @@ contract Propose is AlphaProposalTest {
     // Ensure the governorBravo is now the admin of the timelock
     assertEq(timelock.admin(), address(governorBravo));
   }
+
+  ////
+  // Post proposal tests
+  ////
```
Comment (alexkeating) [inline test/PooltogetherGovernor.t.sol:143]: Take from gitcoin, but reorganized

File: test/PooltogetherGovernor.t.sol [hunk 2]
```diff
@@ -135,4 +137,162 @@ contract Propose is ProposalTest {
...
+
+  function testFuzz_OldGovernorCannotSendETHAfterProposalSucceeds(
+    uint256 _amount,
+    address _receiver
+  ) public {
+    _assumeReceiver(_receiver);
+
+    // Counter-intuitively, the Governor must hold the ETH, not the Timelock.
+    // See the deployed GovernorAlpha, line 227:
+    //   https://etherscan.io/address/0xDbD27635A534A3d3169Ef0498beB56Fb9c937489#code
```
Comment (apbendi) [inline test/PooltogetherGovernor.t.sol]: Needs to be updated

**Review by apbendi — COMMENTED** [review]
One minor note on an outdated comment @alexkeating

File: test/PooltogetherGovernor.t.sol (hunk 2 above)
Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/12/commits/e559c02640193e65b0305cd4d8fd2ae3c7cf27e4#diff-c21c7f9958e9aa1f065be9cab617cf6ffda75c3944410c62779e3eefeb2a2d4cR189

File: test/PooltogetherGovernor.t.sol [hunk 3]
```diff
@@ -135,4 +137,162 @@ contract Propose is ProposalTest {
...
+    assertEq(_receiver.balance, _receiverETHBalance);
+  }
+
+  function testFuzz_OldGovernorSendsTokenAfterProposalIsDefeated(
+    uint256 _amount,
+    address _receiver,
+    uint256 _seed
+  ) public {
+    _assumeReceiver(_receiver);
+    IERC20 _token = _randomERC20Token(_seed);
```
Comment (apbendi) [inline test/PooltogetherGovernor.t.sol:229]: We should add a lot more tokens to this function to test the various holdings of the PoolTogether Timelock. Probably anything over a grand should be included:

Screen Shot 2023-07-18 at 11 03 56 AM

Also note the Timelock holds over $1 M in [PTaUSDC](https://etherscan.io/token/0xdd4d117723c257cee402285d3acf218e9a8236e1), which is a receipt token for USDC deposited in PoolTogether. We definitely want to test this one as well.

**Review by apbendi — COMMENTED** [review]
One more small thing @alexkeating, we should expand our token list. Otherwise this LGTM.

File: test/PooltogetherGovernor.t.sol (hunk 3 above)
Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol:229]: I added a couple tokens. I tried changing the tokens to ADAI and STETH, but ran into issues with some tests. I am looking into it now, but am guessing this is due to how the tokens work.

Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol:229]: For ADAI the balance looks time dependent https://etherscan.io/address/0xc6845a5c768bf8d7681249f8927877efda425baf#code#F26#L57

Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol:229]: I think there is a rounding error with the fuzz tests for STEth. I will add a separate test specifically for STETH

Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol:229]: https://docs.lido.fi/guides/steth-integration-guide#1-2-wei-corner-case

Comment (alexkeating) [reply test/PooltogetherGovernor.t.sol:229]: Added Steth specific tests https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/12/commits/f91c7d38a3d12747833d799b60ec4bf7e39a161f

Add more tokens https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/12/commits/360959be670919f395c7a1155683cff1a93e52f3
